code/modules/holomap/holomap.dm

```diff
@@ -0,0 +1,491 @@
+					neighbor_space = TRUE
+					break
+
+				if(found_area != A || locate(/obj/machinery/door/airlock) in found_turf)
+					neighbors_alien += i
+					continue
+
+				if(found_window ||found_turf.density)
+					neighbors_solid += i
+
+	//-----------------{{{ RENDERING PHASE }}}----------------------------
+
+			if(neighbor_space || length(neighbors_solid) >= 3)
+				if( W || T.density)
+					icon_state_to_use = "full"
+				else if(isfloorturf(T))
+					icon_state_to_use = "floor"
+
+			//--> (not next to space; not surrounded; less than 3 solid neighbors)
+			else if(W || T.density)
+				if(length(neighbors_alien) == 1)
+					icon_state_to_use = "wall"
+					if(length(neighbors_solid) == 1)
+						if((neighbors_alien[1] == NORTH && neighbors_solid[1] == SOUTH) || (neighbors_alien[1] == WEST && neighbors_solid[1] == EAST))
+							dir_to_use = NORTHWEST
+						else if(neighbors_alien[1] == SOUTH && neighbors_solid[1] == NORTH)
+							dir_to_use = SOUTHWEST
+						else if(neighbors_alien[1] == EAST && neighbors_solid[1] == WEST)
+							dir_to_use = NORTHEAST
+						else
+							dir_to_use = neighbors_alien[1]
+					else
+						dir_to_use = neighbors_alien[1]
+				else if(length(neighbors_alien) == 2)
+					icon_state_to_use = "wall"
+					if((neighbors_alien[1] == NORTH && neighbors_alien[2] == SOUTH))
+						icon_state_to_use = "full"
+					else if((neighbors_alien[1] == NORTH && neighbors_alien[2] == EAST))
+						dir_to_use = NORTHEAST
+					else if(neighbors_alien[1] == NORTH && neighbors_alien[2] == WEST)
+						dir_to_use = NORTHWEST
+					else if(neighbors_alien[1] == SOUTH && neighbors_alien[2] == EAST)
+						dir_to_use = SOUTHEAST
+					else if(neighbors_alien[1] == SOUTH && neighbors_alien[2] == WEST)
+						dir_to_use = SOUTHWEST
+				else
+					switch(length(neighbors_solid))
+						if(0)
+							icon_state_to_use =  "full"
+						if(1)
+							if(length(neighbors_alien))
+								icon_state_to_use =  "full"
+							else
+								icon_state_to_use = "wall"
+								if(neighbors_solid[1] == SOUTH || neighbors_solid[1] == NORTH)
+									dir_to_use = WEST
+								else if(neighbors_solid[1] == EAST || neighbors_solid[1] == WEST)
+									dir_to_use = NORTH
+						if(2)
+							if(length(neighbors_alien))
+								icon_state_to_use =  "full"
+							else
+								icon_state_to_use = "wall"
+								if(neighbors_solid[1] == NORTH && neighbors_solid[2] == SOUTH)
+									dir_to_use = WEST
+								else if((neighbors_solid[1] == NORTH && neighbors_solid[2] == EAST))
+									dir_to_use = NORTH
+								else if(neighbors_solid[1] == NORTH && neighbors_solid[2] == WEST)
+									dir_to_use = NORTH
+								else if(neighbors_solid[1] == SOUTH && neighbors_solid[2] == EAST)
+									dir_to_use = NORTHWEST
+								else if(neighbors_solid[1] == SOUTH && neighbors_solid[2] == WEST)
+									dir_to_use = WEST
+								else if(neighbors_solid[1] == EAST && neighbors_solid[2] == WEST)
+									dir_to_use = NORTH
```

Comment:
same here

Reply:
same story, just checking all the different possible scenarios in which two adjacent turfs are solid and what directions they could be in.

Reply:
Thanks for clarifying, it was a little confusing but i see it now